Updated the patch to drop (conflicting) change to README.md. We just want the
code and test fixes here.

From 46cf4ded12ee974a70bf8457263b67ad7ee0379d Mon Sep 17 00:00:00 2001
From: Eddie Gerbais-Nief <eddie.gerbais-nief@proton.me>
Date: Mon, 10 Feb 2025 15:30:20 +0100
Subject: [PATCH] add static to local function in scanner, README update

---
 src/scanner.c            |  4 ++--
 src/tree_sitter/array.h  |  3 ++-
 test/corpus/negative.scm |  8 ++-----
 4 files changed, 29 insertions(+), 34 deletions(-)

diff --git a/src/scanner.c b/src/scanner.c
index e6a32fc..0468564 100644
--- a/src/scanner.c
+++ b/src/scanner.c
@@ -385,7 +385,7 @@ void tree_sitter_typst_external_scanner_deserialize(
 }
 
 // SCAN ////////////////////////////////////////////////////////////////////////
-bool parse_comment(struct scanner* self, TSLexer* lexer) {
+static bool parse_comment(struct scanner* self, TSLexer* lexer) {
 	lex_advance_if(lex_next == '/');
 	if (lex_next == '/') {
 		lex_advance();
@@ -424,7 +424,7 @@ bool parse_comment(struct scanner* self, TSLexer* lexer) {
 	}
 	return false;
 }
-bool parse_space(struct scanner* self, TSLexer* lexer) {
+static bool parse_space(struct scanner* self, TSLexer* lexer) {
 	lex_advance_if(is_sp(lex_next));
 	while (is_sp(lex_next)) {
 		lex_advance();
diff --git a/src/tree_sitter/array.h b/src/tree_sitter/array.h
index 15a3b23..a17a574 100644
--- a/src/tree_sitter/array.h
+++ b/src/tree_sitter/array.h
@@ -14,6 +14,7 @@ extern "C" {
 #include <string.h>
 
 #ifdef _MSC_VER
+#pragma warning(push)
 #pragma warning(disable : 4101)
 #elif defined(__GNUC__) || defined(__clang__)
 #pragma GCC diagnostic push
@@ -278,7 +279,7 @@ static inline void _array__splice(Array *self, size_t element_size,
 #define _compare_int(a, b) ((int)*(a) - (int)(b))
 
 #ifdef _MSC_VER
-#pragma warning(default : 4101)
+#pragma warning(pop)
 #elif defined(__GNUC__) || defined(__clang__)
 #pragma GCC diagnostic pop
 #endif
diff --git a/test/corpus/negative.scm b/test/corpus/negative.scm
index a174353..8a1eef2 100644
--- a/test/corpus/negative.scm
+++ b/test/corpus/negative.scm
@@ -6,9 +6,7 @@ _
 ---------------------
 
 (source_file
-	(emph
-		(item)
-		(MISSING _)))
+	(ERROR))
 
 
 =====================
@@ -20,6 +18,4 @@ negative/001
 ---------------------
 
 (source_file
-	(emph
-		(item)
-		(MISSING _)))
+	(ERROR))
